src/transformers/models/textnet/convert_textnet_original_to_pytorch.py

```diff
@@ -0,0 +1,201 @@
+        ],
+        hidden_sizes=[
+            config_dict["first_conv"]["out_channels"],
+            backbone_config["stage1"]["out_channels"][-1],
+            backbone_config["stage2"]["out_channels"][-1],
+            backbone_config["stage3"]["out_channels"][-1],
+            backbone_config["stage4"]["out_channels"][-1],
+        ],
+        out_features=["stage1", "stage2", "stage3", "stage4"],
+        out_indices=[1, 2, 3, 4],
+        image_size=(size, size),
+    )
+
+    return textnet_config
+
+
+def convert_textnet_checkpoint(checkpoint_url, checkpoint_config_filename, pytorch_dump_folder_path):
+    config_filepath = hf_hub_download(repo_id="Raghavan/fast_model_config_files", filename="fast_model_configs.json")
+
+    with open(config_filepath) as f:
+        content = json.loads(f.read())
+
+    size = content[checkpoint_config_filename]["short_size"]
+
+    if "tiny" in content[checkpoint_config_filename]["config"]:
+        config = prepare_config(tiny_config_url, size)
+    elif "small" in content[checkpoint_config_filename]["config"]:
+        config = prepare_config(small_config_url, size)
+    else:
+        config = prepare_config(base_config_url, size)
+
+    model = TextNetBackbone(config)
+    textnet_image_processor = TextNetImageProcessor(size={"shortest_edge": size})
+    state_dict = torch.hub.load_state_dict_from_url(checkpoint_url, map_location="cpu", check_hash=True)["ema"]
+    state_dict_changed = OrderedDict()
+    for key in state_dict:
+        if "backbone" in key:
+            val = state_dict[key]
+            new_key = key
+            for search, replacement in rename_key_mappings.items():
+                if search in new_key:
+                    new_key = new_key.replace(search, replacement)
+
+            pattern = r"textnet\.stage(\d)"
+
+            def adjust_stage(match):
+                stage_number = int(match.group(1)) - 1
+                return f"textnet.encoder.stages.{stage_number}.stage"
+
+            # Using regex to find and replace the pattern in the string
+            new_key = re.sub(pattern, adjust_stage, new_key)
+            state_dict_changed[new_key] = val
+    model.load_state_dict(state_dict_changed)
+
+    url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+    image = Image.open(requests.get(url, stream=True).raw).convert("RGB")
+
+    original_pixel_values_filepath = hf_hub_download(
+        repo_id="Raghavan/fast_model_samples", filename="original_processed_pixel_values.npy", repo_type="dataset"
+    )
+    original_pixel_values = torch.from_numpy(np.load(original_pixel_values_filepath))
+    pixel_values = textnet_image_processor(image, return_tensors="pt").pixel_values
+
+    assert torch.allclose(original_pixel_values, pixel_values)
+
+    with torch.no_grad():
+        output = model(pixel_values)
+
+    assert np.allclose(output["feature_maps"][-1][0][0][0][-2:].detach().numpy(), np.array([4.0259, 17.4911]))
```

Comment:
Just to confirm these values were derived from passing the pixel values to the original model? Could you add an assert message to show the expected behaviour (and effectively document this?). 

It's be good to test on a few more values than just 2